package-res/resources/web/common-ui-require-js-cfg.js

```diff
@@ -63,6 +64,14 @@
   requirePaths["pentaho/_globalContextVars"] = basePath + "/pentaho/_globalContextVars";
   requirePaths["pentaho/contextVars"] = requirePaths["pentaho/_globalContextVars"];
 
+  // Singleton interface, virtual module ids
+  //requireMap["*"]["pentaho/spec/IContext"] = "pentaho/service!pentaho.spec.IContext?single";
+  requireMap["*"]["pentaho/config/IService"] = "pentaho/service!pentaho.config.IService?single";
+
+  // Default implementations
+  requireService["pentaho/config/impl/serviceInstance"] = "pentaho.config.IService";
+  requireService["pentaho/config/impl/AmdLoadedService"] = "Class<pentaho.config.IService>";
+
```

Comment:
I dislike the use of `api` — the regular namespaces _are the API_. The API side can contain interfaces, mandatory base classes (abstract or not), enums.
A sub-folder named `impl`, anywhere in a regular namespace, would contain auxiliary types, public and/or private, that support implementations of API types.

Ultimately, the impl subfolders could be split into a separate source root. This wouldn't map to a tree without "impl" subfolders, like would happen if "impl" could be implied... Could be used, however, to evidence separation and make it easier to distribute as separate bundles...